tests/test_cli.py

```diff
@@ -76,3 +76,20 @@ def print_helper(_path):
 
     with pytest.raises(NotADirectoryError, match=print_helper(path)):
         assert_valid_structure(TEST_DATA_DIR / "invalid_yaml")
+
+def test_cli_custom_dimensions_quotes():
+    """Check that CLI expected simple quotes (') inside double quotes (")"""
+    result_valid = runner.invoke(
+        cli, ["validate-project", 
+            str(TEST_DATA_DIR / "non-default_dimensions_passing"), 
+            "--dimensions \"['variable', 'region', 'foo']\""
```

Comment:
Depending on how click constructs the argument, I think this might fix your issue:
```suggestion
            "--dimensions", "\"['variable', 'region', 'foo']\""
```
i.e. having `--dimensions` and the value of the `dimensions` option as two different string entries in the list of commands.

Reply:
Unfortunately the click documentation is not really detailed (https://click.palletsprojects.com/en/8.1.x/testing/) but I would imagine that it works in analogy to how the python standard library does it when you invoke the command line from python (https://docs.python.org/3/library/subprocess.html#popen-constructor). 
In the example given in the link, in order to execute:
```console
/usr/bin/git commit -m "Fixes a bug."
```
from a python script you'd have to use:
```python 
Popen(["/usr/bin/git", "commit", "-m", "Fixes a bug."])
```
The important point is that `-m ` and `"Fixes a bug"` are two separate entries in the list. I think you'll have to do this in analogy to your `--dimensions` and `['variable', ...]`.

Reply:
OK thank you! I does not work yet but I will try :)